src/integrationTest/sam.test.ts

```diff
@@ -397,246 +397,241 @@ describe('SAM Integration Tests', async function () {
         console.log(`DebugSessions seen in this run:\n${sessionReport}`)
     })
 
-    for (let scenarioIndex = 0; scenarioIndex < scenarios.length; scenarioIndex++) {
-        const scenario = scenarios[scenarioIndex]
-
-        describe(`SAM runtime: ${scenario.displayName}`, async function () {
-            let runtimeTestRoot: string
-
-            before(async function () {
-                runtimeTestRoot = path.join(testSuiteRoot, scenario.runtime)
-                console.log('runtimeTestRoot: ', runtimeTestRoot)
-                mkdirpSync(runtimeTestRoot)
-            })
-
-            after(async function () {
-                // don't clean up after java tests so the java language server doesn't freak out
-                if (scenario.language !== 'java') {
-                    await tryRemoveFolder(runtimeTestRoot)
-                }
-            })
-
-            function log(o: any) {
-                console.log(`sam.test.ts: scenario ${scenarioIndex} (${scenario.displayName}): ${o}`)
-            }
-
-            /**
-             * This suite cleans up at the end of each test.
-             */
-            describe('Starting from scratch', async function () {
-                let testDir: string
+    function uniqueTestDirPath(rootPath: string): string {
+        return path.join(rootPath, `scenario-${randomUUID()}`)
```

Comment:
these tests are not parallelized, but even if they were, it should be possible to increment an integer to get a unique name (which is shorter / more human-readable in logs). (The before() step should of course delete the dir first, if it exists).